### PR TITLE
Update IDL patch for Web Animations

### PR DESCRIPTION
**Pull request by tidoust**
Patch is still needed but IDL right before the patch got changed in the spec invalidating the previous patch.

### CODE REVIEW

**Comment by dontcallmedom** [comment]
The said recent changes in web-animations-2 redefine the types of existing elements in `ComputedEffectTiming` which creates overlapping IDL definitions that makes the test fail.

I guess we need to add removing these duplicates in the patch?

**Comment by tidoust** [comment]
Yes :( Will do.

File: ed/idlpatches/web-animations-2.idl.patch [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
 --- a/ed/idl/web-animations-2.idl
 +++ b/ed/idl/web-animations-2.idl
-@@ -38,8 +38,6 @@ partial dictionary ComputedEffectTiming {
-     CSSNumberish?        progress;
+@@ -30,16 +30,6 @@ partial dictionary OptionalEffectTiming {
+     double playbackRate;
  };
  
+-partial dictionary ComputedEffectTiming {
+-    CSSNumberish         startTime;
```
Comment (dontcallmedom) [inline ed/idlpatches/web-animations-2.idl.patch:22]: sorry, I think we still need to keep `startTime` which only appears in v2 AFAICT

Comment (tidoust) [reply ed/idlpatches/web-animations-2.idl.patch:22]: True

Comment (tidoust) [reply ed/idlpatches/web-animations-2.idl.patch:22]: The type of all other properties also change in that dfn to `CSSNumberish`. I'm wondering whether it wouldn't be preferable to add a patch for Web Animations Level 1 that drops `ComputedEffectTiming`?

Comment (dontcallmedom) [reply ed/idlpatches/web-animations-2.idl.patch:22]: I wondered about it too; but then we would still need a patch to make `ComputedEffectTiming` no longer a *partial* dictionary. I don't have strong feelings (except that delta specs are a pain)

Comment (tidoust) [reply ed/idlpatches/web-animations-2.idl.patch]: I did that. Feeling adventurous today...

**Review by dontcallmedom — APPROVED** [review]

**Comment by saschanaz** [comment]
Do we have an upstream tracking bug for this?

**Comment by dontcallmedom** [comment]
we didn't think we could raise one given that web-animation-2 is explicitly marked as a delta spec (where this kind of overlap is to be expected); but maybe the partial declaration of the dictionary should be flagged?

File: ed/idlpatches/web-animations-2.idl.patch [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
 --- a/ed/idl/web-animations-2.idl
 +++ b/ed/idl/web-animations-2.idl
-@@ -33,8 +33,6 @@ partial dictionary ComputedEffectTiming {
-     double startTime;
+@@ -30,7 +30,7 @@ partial dictionary OptionalEffectTiming {
+     double playbackRate;
+ };
+ 
+-partial dictionary ComputedEffectTiming {
++dictionary ComputedEffectTiming {
```
Comment (saschanaz) [inline ed/idlpatches/web-animations-2.idl.patch:25]: This needs `: EffectTiming`, right? I don't see why not.

Comment (dontcallmedom) [reply ed/idlpatches/web-animations-2.idl.patch:25]: you're absolutely right, we missed that completely.

**Pull request merged**